src/main/java/org/black_ixx/bossshop/core/BSShop.java

```diff
@@ -89,8 +90,9 @@ public void setDisplayName(String displayname) {
 
     public String getValidDisplayName(Player p, BSShopHolder holder) {
         String displayname = this.displayname;
+        String displayname2 = ChatColor.stripColor(displayname);
         displayname = ClassManager.manager.getStringManager().transform(displayname, null, this, holder, p);
-        return displayname.length() > 40 ? displayname.substring(0, 40) : displayname;
+        return displayname2.length() > 45 ? displayname.substring(0, 45) : displayname;
```

Comment:
I appreciate what you're trying to do here, but if the display name can only be 45 characters, color codes included, we shouldn't be checking the stripped version to determine length.

Reply:
I still don't quite understand what you mean

Reply:
> I appreciate what you're trying to do here, but if the display name can only be 45 characters, color codes included, we shouldn't be checking the stripped version to determine length.

As far as I know, the color codes should be excluded from the count as per [this conversation](https://github.com/promcteam/BossShopPro/issues/12). If anything, the character limitation should be removed altogether. Is there a limit on chest titles as of now?

It doesn't seem to be checked for within the CraftServer class:
`    public Inventory createInventory(InventoryHolder owner, int size, String title) throws IllegalArgumentException {
        Validate.isTrue(9 <= size && size <= 54 && size % 9 == 0, "Size for custom inventory must be a multiple of 9 between 9 and 54 slots (got " + size + ")");
        return CraftInventoryCreator.INSTANCE.createInventory(owner, size, title);
    }`

Reply:
Ah that's fair. My bad. I'm still not sure where the 45 number is coming from either. I'd be more inclined to drop the character limit entirely. Afaik, the 32 char limit was dropped in 1.13. I'm not entirely sure though. At least in later versions it's not there to my knowledge.

Maybe the solution here is to enforce a character limit on versions before 1.15 and don't worry about it on later versions.

Reply:
I also recall the 32-char limit being dropped in 1.13. Let's keep it out! 😊